consensus/event_loop.go

```diff
@@ -109,41 +105,34 @@ func (fc *FastHotStuffEventLoop) AdvanceView() (uint64, error) {
 	return fc.currentView, nil
 }
 
-// ProcessSafeBlock must only be called when the caller has accepted a new block, connected it
+// ProcessTipBlock must only be called when the caller has accepted a new block, connected it
 // to the tip of the blockchain, and determined that the block is safe to vote on. Given such a
 // block, this function resets internal state and schedules the next block construction and timeout
-// determine the next action. The functions expects the following for the input params:
-//   - block: the input block that was safely added to the blockchain and is safe to vote on
-//   - validators: the validator set for the next block height
-func (fc *FastHotStuffEventLoop) ProcessSafeBlock(block Block, validators []Validator) error {
+// timers.
+//
+// Expected params:
+//   - tip: the current uncommitted tip of the blockchain, with the validator set at that block height
+//   - safeBlocks: an unordered slice of blocks including the committed tip, the uncommitted tip,
+//     all ancestors of the uncommitted tip that are safe to extend from, and all blocks from forks
+//     that are safe to extend from. This function does not validate the collection of blocks. It
+//     expects the caller to know and decide what blocks are safe to extend from.
+func (fc *FastHotStuffEventLoop) ProcessTipBlock(tip BlockWithValidators, safeBlocks []BlockWithValidators) error {
```

Comment:
Seems like a more accurate name for this function would be `ProcessFork` or something alike. Since it's basically what we're doing here.

Reply:
This function would be called whenever the tip changes. Which could happen due to a fork, or when the previous tip is extended. Can't say if any of these are any more representative of what it does: ProcessFork, ProcessNewTip,...

FWIW, this function has had many names in the past (UpdateChainTip, ProcessSafeBlock, ProcessTipBlock,...). But they all did the same thing 😄. So I'm more inclined to just run with the current name for now

Reply:
Haha you're right we keep changing it. Alright let's roll with ProcessTipBlock